### PR TITLE
Make css dependency graph of js modules

### PR DESCRIPTION
**Pull request by jasongrout**
## References

This explores implementing https://github.com/jupyterlab/jupyterlab/issues/9375#issuecomment-733193129

This depends on https://github.com/jupyterlab/lumino/pull/136, which is a companion PR that adds similar style/index.js files to Lumino. This has been merged and released.

This approach avoids the problems with css-loader not deduplicating css noted in https://github.com/jupyterlab/jupyterlab/pull/9364

## Code changes

This introduces a dependency graph for css using a new convention, `style/index.js` files. This way the dependency tree is in js, so it takes advantage of all of the webpack bundling, module federation sharing, etc., but doesn't hardcode a css import in the main js file. At the top application level, instead of generating an import.css with all the css imports, we generate a style.js file.

With this change, we are able to upgrade `css-loader` and still maintain about 100 style tags on the page with no duplicates that I saw. 

Overall, I think deduplicating the actual css at runtime (https://github.com/jupyterlab/jupyterlab/issues/9375#issuecomment-736867083) is probably a better thing to do in the end, but I'm not sure how straightforward that approach will be or how long it will take, so I opted to first do this approach for 3.0, where I could pretty clearly see what to do.

### CODE REVIEW

**Comment by jasongrout** [comment]
I updated the css-loader, and confirmed that we still only have about 100 style tags on the page (now that we are not relying on css-loader to do deduplication). Whether this holds out for lots of extensions is to be determined - we may still need to further automate a solution for extensions, for css-loader to do deduplication at the page level at runtime, or revert back to our old version of css-loader.

**Comment by jasongrout** [comment]
I rebased and cleaned up/combined commits.

**Comment by jasongrout** [comment]
I think this is ready for review at this point. After review, it would be great to get another RC with this released.

File: builder/src/build.ts [hunk 1]
```diff
@@ -233,23 +225,47 @@ export namespace Build {
...
+      }
+    }
+
+    let jsImport;
     cssImports.forEach(cssImport => {
-      cssContents += `\n@import url('~${cssImport}');`;
+      if ((jsImport = cssJS(cssImport))) {
+        styleContents.push(`import '${jsImport}';`);
+      } else {
+        styleContents.push(`// import '${cssImport}';`);
```
Comment (jasongrout) [inline builder/src/build.ts:256]:
```suggestion
        styleContents.push(`import '${cssImport}';`);
```
Import css when we need to, just like in ensure-repo

**Review by blink1073 — APPROVED** [review]
LGTM, thanks (pending the suggested change above)!

**Comment by cceyda** [comment]
I'm getting mathjax error with the jlab 3.0.0rc11
image

which makes the cells become weird:
image

This is probably related to some commit that is after 3d81e1ae214ff0c9591fc31772f48093a9635e05 (which didn't have this problem)

**Comment by jasongrout** [comment]
Indeed, I'm also seeing very strange things happening with notebook cells in 3.0rc11. I'll open a new issue.

**Comment by jasongrout** [comment]
I opened #9423 about the issue, and fixed the underlying issue in #9427.

**Comment by jasongrout** [comment]
@cceyda - 3.0rc12 was released just now fixing those codemirror issues. Thanks again.

**Pull request merged**